packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:file/memory.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/tools/asset_transformer.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
+
+import '../../../src/common.dart';
+import '../../../src/fake_process_manager.dart';
+
+void main() {
+  test('Invokes dart properly', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync()..writeAsStringSync('hello world');
+    const String outputPath = 'output.txt';
+
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          artifacts.getArtifactPath(Artifact.engineDartBinary),
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
```

Comment:
if you make these deterministic, please update this to be a const string. i want to make sure we don't break this argument :)

Reply:
Yeah, I think making these deterministic would make these tests much easier to maintain. I may or may not have spent a good hour figuring out regex mis-matches in some the later asserts in this file 🙃

Reply:
One downside of using const strings here would be that we are depending on the implementation of `MemoryFileSystem::systemTempDirectory`.

Reply:
I can live with that: https://github.com/google/file.dart/blob/master/packages/file/lib/src/backends/memory/memory_file_system.dart#L125

Reply:
If this changes, it'll break presubmits on a pub roll. And here's the changelog of that particular library: https://github.com/google/file.dart/commits/master/packages/file/lib/src/backends/memory/memory_file_system.dart

Reply:
Ack